File: 구은성/9월/0921/87389.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        if (isPrime(n - 1)) {
            answer = n - 1;
        }
        else {
            answer = minDivisor(n - 1);
        }
        
        return answer;
    }
    
    public boolean isPrime(int n) {
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    public int minDivisor(int n) {
        int minDiv = n + 1;
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0 && i < minDiv) {
                minDiv = i;
            }
        }
        return minDiv;
    }
}

File: 정여진/10월/2주차/1009/87389.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    for (int i=2;i<=n+1;i++) {
        if (n%i==1) {
            answer=i ;
            break ;
        }
    }
    return answer;
}